### PR TITLE
Change/rebuild ally

### PR DESCRIPTION
**Pull request by karlthepagan**
p.s. my spawns are gone and I need a rebuild

### CODE REVIEW

File: room.js [hunk 1]
```diff
@@ -532,7 +532,7 @@ mod.extend = function(){
             configurable: true,
             get: function() {
                 if( _.isUndefined(this._constructionSites) ) {
-                    this._constructionSites = this.find(FIND_MY_CONSTRUCTION_SITES);
+                    this._constructionSites = this.find(FIND_CONSTRUCTION_SITES);
```
Comment (chobobobo) [inline room.js]: Need to consider any possible side-effects in the code from this change, eg. If I am taking over a room with buildings left over from someone else, will my workers start building those buildings?

File: task.reputation.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ const CONST = {
 
 const mod = {
     name: 'reputation',
-    myName: () => _.values(Game.spawns)[0].owner.username,
+    myName: () => _.chain(Game.rooms).map('controller').flatten().filter('my').map('owner.username').first().value(),
```
Comment (chobobobo) [inline task.reputation.js]: This change is already in #956 will that cause conflicts?

File: room.js (hunk 1 above)
Comment (karlthepagan) [inline room.js]: I agree. This is problematic.

Comment (karlthepagan) [inline room.js]: Possibly resolved by the ally checking in building action.

File: task.reputation.js (hunk 2 above)
Comment (karlthepagan) [inline task.reputation.js]: No, same branch because I'm taking a break and my dev discipline broke down.

File: room.js (hunk 1 above)
Comment (chobobobo) [inline room.js]: I think perhaps adding myConstructionSites which is simply a filter to this.constructionSites would work.. I noticed that miner just does a check on constructionSites, upgrader looks at constructionSites.length, worker spawns are based on the same, getBestConstructionSiteFor also looks at them all.. perhaps a good change here would be to simply take FIND_CONSTRUCTION_SITES and filter for ally owner?

**Comment by chobobobo** [comment]
Ok, added room.myConstructionSites and converted all existing usage.

**Comment by chobobobo** [comment]
This was merged into major/traveler to get some further testing in canary given it will be merged soon.

**Pull request closed without merging**